source/server/admin/config_dump_handler.cc

```diff
@@ -384,10 +384,12 @@ void ConfigDumpHandler::addLbEndpoint(
   endpoint.set_hostname(host->hostname());
   Network::Utility::addressToProtobufAddress(*host->address(), *endpoint.mutable_address());
   if (host->addressListOrNull() != nullptr) {
-    for (auto& additional_address : *host->addressListOrNull()) {
-      auto& new_address = *endpoint.mutable_additional_addresses()->Add();
-      Network::Utility::addressToProtobufAddress(*additional_address,
-                                                 *new_address.mutable_address());
+    const auto& address_list = *host->addressListOrNull();
```

Comment:
Is there a test validating this config-dump change?

Reply:
Yes we have a check on the ouput of the configdump handler https://github.com/envoyproxy/envoy/blob/32318657d912793048339fb392d30dc363a7acac/test/server/admin/config_dump_handler_test.cc#L186-L191

The reason why the test did not fail in the past was because it was written as part of my old PR which assumed address list to contains "additional addresses" only. Both the old PR and the related tests went along with that mistake.
In this PR the existing test has been adjusted to align with the idea of the address list being default_addr+additional_addrs to reflect the new changes.